src/Microsoft.TestPlatform.VsTestConsole.TranslationLayer/VsTestConsoleWrapper.cs

```diff
@@ -172,6 +173,73 @@ public void StartSession()
             }
         }
 
+        /// <inheritdoc/>
+        [Obsolete("This API is not final yet and is subject to changes.", false)]
+        public ITestSession StartTestSession(
+            IList<string> sources,
+            string runSettings,
+            ITestSessionEventsHandler eventsHandler)
+        {
+            return this.StartTestSession(
+                sources,
+                runSettings,
+                options: null,
+                eventsHandler);
+        }
+
+        /// <inheritdoc/>
+        [Obsolete("This API is not final yet and is subject to changes.", false)]
+        public ITestSession StartTestSession(
+            IList<string> sources,
+            string runSettings,
+            TestPlatformOptions options,
+            ITestSessionEventsHandler eventsHandler)
+        {
+            return this.StartTestSession(
+                sources,
+                runSettings,
+                options,
+                eventsHandler,
+                testHostLauncher: null);
+        }
+
+        /// <inheritdoc/>
+        [Obsolete("This API is not final yet and is subject to changes.", false)]
+        public ITestSession StartTestSession(
+            IList<string> sources,
+            string runSettings,
+            TestPlatformOptions options,
+            ITestSessionEventsHandler eventsHandler,
+            ITestHostLauncher testHostLauncher)
+        {
+            this.testPlatformEventSource.TranslationLayerStartTestSessionStart();
```

Comment:
I'm wondering why the sync APIs don't delegate to their async equivalents.
For instance `StartTestSession(params) => StartTestSessionAsync(params).Result`.

This would ensure that there's just one code path for sync and async APIs and it might become easier to stabilize the async APIs this way.

Reply:
Totally agree with this remark. However, the reason I implemented it this way is to mirror the implementations of async APIs for discovery/run operations. I know there are some cases in which the async APIs don't work as expected and until we iron those out we cannot delegate the call to the async APIs just yet. The scope of such a change would be quite large anyways and this PR is already large enough as it is, so maybe it'd be better to address this in a separate PR.